### PR TITLE
IN-49 Added automated testing and linting support

### PR DESCRIPTION
**Pull request by fork-boy**
## Description
Imported jest package for automated testing. Added script to run jest inside package JSON. Perform the test on the Home component which is inside the index file and created a test component inside __test__ folder and it ran successfully and shows the output as expected. Added linting support, the `npm run lint` command shows all the linting errors, and the `npm run lint: fix` command fixes all the errors that were shown.

## Screenshot
Screenshot at 2021-04-13 13-38-11

## Tests
 The manual test was done and an automated test was done on the Home component

### Automated test cases added
- Automated test done to render the expected output of Home component using jest

### Manual test cases run
- `npm run test` command run successfully and shows the expected pass output

### Browsers tested
- [ ] Chrome
- [ ] Safari
- [ ] Internet Explorer
- [ ] Firefox

## Platform dependencies
Does this depend on or impact
- [ ] iOS
- [ ] Android
- [ ] API

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - checkout
+      - run:
+          name: Update NPM
+          command: 'sudo npm install -g npm@5'
+      - restore_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+      - run:
+          name: Install Dependencies
+          command: npm install
```
Comment (jjalan) [inline .circleci/config.yml]: I think you want to do `npm ci`. Please check.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - save_cache:
+          key: dependency-cache-{{ checksum "package-lock.json" }}
+          paths:
+            - ./node_modules
+      - run:
+          name: Run tests
```
Comment (jjalan) [inline .circleci/config.yml:25]: Should we also be running lint check?

File: package.json [hunk 3]
```diff
@@ -5,16 +5,40 @@
   "scripts": {
     "dev": "next dev",
     "build": "next build",
-    "start": "next start"
+    "start": "next start",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:coverage": "jest --coverage",
+    "lint": "eslint --fix ./pages/**/*.ts[x]"
```
Comment (jjalan) [inline package.json]: --fix will autofix.

We do not want to auto fix in CircleCI. It will make the error.

**Review by jjalan — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (fork-boy) [reply .circleci/config.yml]: ok

File: package.json (hunk 3 above)
Comment (fork-boy) [reply package.json]: ok

File: .circleci/config.yml (hunk 2 above)
Comment (fork-boy) [reply .circleci/config.yml:25]: yes sir, adding that

**Review by jjalan — APPROVED** [review]

**Pull request merged**